poll.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/pborman/getopt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+var STATUSES = [5]string{"UNKNOWN", "DEPLOYED", "DELETED", "SUPERSEDED", "FAILED"}
+
+const NUM_MOCKED_RELEASES = "NUM_MOCKED_RELEASES"
+const DEPLOYED_STATUS = "DEPLOYED"
+const INSTALLING_STATUS = "INSTALLING"
+const RELEASE_STATUSES = "RELEASE_STATUSES"
+
+type Runner interface {
+	Run(string, ...string) (string, error)
+}
+
+type RealRunner struct{}
+
+var runner Runner
```

Comment:
This was the only way in which i could mock a command line runner for the tests. I did try gomock but started running into a lot of issues quickly.
I would also prefer to avoid using global vars. Do you have any suggestion?

Reply:
You could instantiate the runner on `main()` and pass it as argument to the methods.
The same for the tests, in which you would instantiate the object and pass it to the methods. Like this:

```
runner := TestRunner{1, INSTALLING_STATUS}
out := PollRelease(runner, "fakerelease", 10, 10)
```

What do you think?

Reply:
Of course!!!
Injecting it into the method. Thanks! I am refactoring this now!